### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by kelmag**
In this PR I migrated to a new API @smakosh helped me with, I was working on the API to adapt it to the spec I have on this app, this PR is mainly about : 

1. Migrate the login and the post forms to Formik with Yup

### CODE REVIEW

File: src/components/TextButton/index.js [hunk 1]
```diff
@@ -1,66 +1,50 @@
...
-  const {
-    customStyle,
-    disabledAuthor,
-    button,
-    rows,
-    handleClick,
-    setTextValue,
-    textValue
-  } = props;
+  const { customStyle, disabledAuthor, button, fieldComponent, rows } = props;
```
Comment (smakosh) [inline src/components/TextButton/index.js]: you could do destructure these props from above, ex:
```
const TextButton = ({ customStyle, disabledAuthor, button, fieldComponent, rows }) => (
...
)
```

File: src/components/TextButton/index.js [hunk 2]
```diff
@@ -1,66 +1,50 @@
...
+const FormikApp = withFormik({
+  mapPropsToValues({ content }) {
+    return {
+      content: content || ""
+    };
+  },
+  validationSchema: Yup.object().shape({
+    content: Yup.string().required()
+  }),
+  handleSubmit(values, { props, resetForm, setSubmitting }) {
```
Comment (smakosh) [inline src/components/TextButton/index.js]: `handleSubmit({ content }, { props: { onSubmit }, resetform, setSubmitting }) { ...`

**Comment by smakosh** [comment]
I'm still not getting automatically logged in even though my token is stored on localStorage

**Comment by smakosh** [comment]
Also add Netlify redirects to redirect all routes to trigger the index.html file:

[About redirects](https://www.netlify.com/docs/redirects/)

TL;DR:
1- create a file called `_redirects` inside `public/` folder
2- paste this in `/*    /index.html   200`
